lib/project_types/extension/models/development_server.rb

```diff
@@ -18,18 +18,46 @@ def create(server_config)
       end
 
       def build(server_config)
-        _, error, pid = CLI::Kit::System.capture3(executable, "build", "-", stdin_data: server_config.to_yaml)
-        return if pid.success?
+        _, error, status = CLI::Kit::System.capture3(executable, "build", "-", stdin_data: server_config.to_yaml)
+        return if status.success?
         raise DevelopmentServerError, error
       end
 
-      def serve
-        raise NotImplementedError
+      def serve(context, server_config)
+        CLI::Kit::System.popen3(executable, "serve", "-") do |input, out, err, status|
+          context.puts("Sending configuration data …")
+          input << server_config.to_yaml
+          input.close
+
+          open_threads(out, err, context)
+
+          status.value
+        end
       end
 
       def version
         raise NotImplementedError
       end
+
+      private
+
+      def open_threads(out, err, ctx)
```

Comment:
Can you rename this to `forward_output_to_user`? I think that captures the intension better instead of describing how this is acomplished with the code itself is doing.